File: 2580-circular-sentence/circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        l = 0
        r = 1
        while l < r < len(sentence):
            while l < r < len(sentence) and sentence[r - 1] != " ":
                r += 1
            while  l < r < len(sentence) and  sentence[l + 1] != " ":
                l += 1
            if l < r < len(sentence) and sentence[r] != sentence[l]:
                return False
            r+=1
            l+=1
        return True if sentence[0] == sentence[-1] else False
            

File: 2580-circular-sentence/circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        int i = 0;
        while (i < sentence.length()){
            if (sentence[i] == ' ' && sentence[i - 1] != sentence[i + 1]) return false;
            i++;
        }
        return sentence[0] == sentence[sentence.length() - 1];
    }
};